routing_plugin.go

```diff
@@ -106,3 +123,522 @@ func ReleaseRoutingPlugin(ctx context.Context) {
 		routingPluginInstance = nil
 	}
 }
+
+// lowToHighRoutingPlugin is a RoutingPlugin that implements "low to high"
+// routing. This means that when we're attempting to pay to a target we'll
+// gradually (with a linear step function) discard inbound peers to that target
+// given routing timeouts. The lowToHighRoutingPlugin itself is responsible for
+// manipulating LND's Mission Control to make such routing attempts possible.
+type lowToHighRoutingPlugin struct {
+	lnd     lndclient.LndServices
+	clock   clock.Clock
+	target  route.Vertex
+	amount  btcutil.Amount
+	mcState map[route.Vertex]lndclient.MissionControlEntry
+
+	// nodesByMaxFee holds nodes sorted by maximum fees that would be paid
+	// to the target node for the target amount.
+	nodesByMaxFee []nodeFeeInfo
+
+	// mcChanged flags that the MC settings for the tracked nodes were
+	// changed and should be reset to their original state once the plugin
+	// is done.
+	mcChanged bool
+}
+
+type nodeFeeInfo struct {
+	node     route.Vertex
+	capacity btcutil.Amount
+	fee      int64
+}
+
+// Enforce that lowToHighRoutingPlugin implements the RoutingPlugin interface.
+var _ RoutingPlugin = (*lowToHighRoutingPlugin)(nil)
+
+// buildPrivateChannels creates the private channel map from the passed route
+// hints. The code is taken and adapted from LND. Original source:
+// lnd/routing/payment_session_source.go.
+func buildPrivateChannels(routeHints [][]zpay32.HopHint,
```

Comment:
So I guess the key difference here is that you want to know all the edges by given a node, rather than having only the directional edges where the node is pointing to next.